tests/rng_test.cpp

```diff
@@ -0,0 +1,63 @@
+#include "catch/catch.hpp"
+
+#include "test_statistics.h"
+
+#include "rng.h"
+
+static void check_remainder( float proportion )
+{
+    statistics<float> stats;
+    const epsilon_threshold confidence_level{ proportion, 0.05 };
+    do {
+        stats.add( roll_remainder( proportion ) );
+    } while( stats.n() < 100 || stats.uncertain_about( confidence_level ) );
+    INFO( "Goal: " << proportion );
+    INFO( "Result: " << stats.avg() );
+    INFO( "Samples: " << stats.n() );
+    CHECK( stats.test_threshold( confidence_level ) );
+}
+
+TEST_CASE( "roll_remainder_distribution" )
+{
+    check_remainder( 0.0 );
+    check_remainder( 0.01 );
+    check_remainder( -0.01 );
+    check_remainder( 1.5 );
+    check_remainder( -1.5 );
+    check_remainder( 1.75 );
+    check_remainder( -1.75 );
+    check_remainder( 1.1 );
+    check_remainder( -1.1 );
+    check_remainder( 1.0 );
+    check_remainder( -1.0 );
+    check_remainder( 10.0 );
+    check_remainder( -10.0 );
+    check_remainder( 10.5 );
+    check_remainder( -10.5 );
+}
+
+static void check_x_in_y( double x, double y )
+{
+    statistics<bool> stats( Z99_99 );
+    const epsilon_threshold confidence_level{ x / y, 0.05 };
+    do {
+        stats.add( x_in_y( x, y ) );
+    } while( stats.n() < 100 || stats.uncertain_about( confidence_level ) );
+    INFO( "Goal: " << x << " / " << y << " ( " << x / y << " )" );
+    INFO( "Result: " << stats.avg() );
+    INFO( "Samples: " << stats.n() );
+    CHECK( stats.test_threshold( confidence_level ) );
```

Comment:
The fact that this [failed](https://travis-ci.org/CleverRaven/Cataclysm-DDA/jobs/528380699) at the 99.99% confidence level is a little concerning.  Wondering whether the stats are in fact being computed correctly.

Reply:
Actually, no, I didn't appreciate just how many tests are being performed here.  It's probably just failing by chance.  So the confidence level should be increased, probably to 99.9999% or so.

Reply:
Gah, I ran this a LOT locally before pushing, but just not enough I guess.

Side note, if you have any thoughts on a better susbet of conditions to check or more meaningful assertions to make, I'm all ears.

One option is hacking up the RNG itself, because I'm implicitly trusting that the std generators we're using are correct, I'm just trying to test our logic on top of them.